### ISSUE TITLE
How to run LLama-3 or Phi with more then 4096 prompt tokens?

### ISSUE THREAD

**Issue by baleksey** [huggingface/candle#2171]
Could you please show me an example where LLama-3 model used (better GGUF quantized) and initial prompt is more then 4096 tokens long? Or better 16-64K long (for RAG). Currently everything I do ends with error:
In this code:
let logits = model.forward(&input, 0);  // input is > 4096 tokens

Error:
narrow invalid args start + len > dim_len: [4096, 64], dim: 0, start: 0, len:4240

Thank you a lot in advance!
